code/modules/modular_computers/os/ui.dm

```diff
@@ -74,14 +74,14 @@
 		return TOPIC_REFRESH
 	if(href_list["PC_enable_component"] )
 		var/obj/item/stock_parts/computer/H = locate(href_list["PC_enable_component"]) in holder
-		if(H && istype(H) && !H.enabled)
-			H.enabled = 1
+		if(H && istype(H) && !(H.status & PART_STAT_ACTIVE))
+			H.set_status(holder, PART_STAT_ACTIVE)
 			H.on_enable(src)
 		return TOPIC_REFRESH
 	if(href_list["PC_disable_component"] )
 		var/obj/item/stock_parts/computer/H = locate(href_list["PC_disable_component"]) in holder
-		if(H && istype(H) && H.enabled)
-			H.enabled = 0
+		if(H && istype(H) && (H.status & PART_STAT_ACTIVE))
+			H.unset_status(holder, PART_STAT_ACTIVE)
```

Comment:
I believe the overall structure of the code is such that you aren't allowed to call this externally on parts: the part calls this internally to trigger events on the machine, but is updating its own status alone. This relates to one of the points in your PR header, that `PART_STAT_ACTIVE` is used inconsistently. For power components, `ACTIVE` is not really "turned on," it's more like "I'm the one currently providing power," and to set/unset it you are supposed to use `not_needed`/`can_provide_power`.

I'd suggest something like a stock part proc for "activated/deactivated by computer." and a getter for "am I on." Then parts can do whatever internally, possibly using a separate flag. It might even be that toggling the `INSTALLED` flag is closer to what you want. Obviously this is annoying to implement until the part types are combined.

I guess it's also worth considering: what purpose does manually activating/deactivating parts without removal in a computer serve? With machines, my conclusion was that it served no purpose, so they just don't have this concept. Of course computers are likely different.